### PR TITLE
Add cli

### PR DESCRIPTION
**Pull request by nbirillo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,3 +21,50 @@ run `./gradlew addKtlintFormatGitPreCommitHook`. As the result, the `.git` folde
...
+
+Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-transformations`.
+
+
+## Getting started as a tool
+
+Run the command `./gradlew :cli -Pinput=<Input directory with python files> -Poutput=<Output directory> -Pyaml=<YAML config path>`.
+The YAML configuration file example can be found [here](./config.yaml).
+Set of the transformations is applied `n transformations` times. Each round for each transformation the random  
+number between `0.0` and `1.0` is generated. If this value less than `p` for the current transformation,
```
Comment (elena-lyulina) [inline README.md:35]: is less

File: README.md [hunk 2]
```diff
@@ -33,3 +21,50 @@ run `./gradlew addKtlintFormatGitPreCommitHook`. As the result, the `.git` folde
...
+the transformation will be applied. The result for each round is stored in the separated folder.
+
+
+## Getting started as a library
+
+Import the plugin in your plugin:
```
Comment (elena-lyulina) [inline README.md:41]: Sounds a little bit strange, maybe somehow rephrase it? For example, "You can use this plugin as a library in your plugin by importing it in `settings.gradle.kts` and `build.gradle.kts` files"

File: README.md [hunk 3]
```diff
@@ -33,3 +21,50 @@ run `./gradlew addKtlintFormatGitPreCommitHook`. As the result, the `.git` folde
...
+## Installation
+
+Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-transformations`.
+
+
+## Getting started as a tool
+
+Run the command `./gradlew :cli -Pinput=<Input directory with python files> -Poutput=<Output directory> -Pyaml=<YAML config path>`.
+The YAML configuration file example can be found [here](./config.yaml).
+Set of the transformations is applied `n transformations` times. Each round for each transformation the random
```
Comment (elena-lyulina) [inline README.md:34]: Maybe rename `n transformations` parameter? Now it looks like the number of transformations, while it's about the number of times set of the transformations should be applied. How about `n apply` or `n times`?

File: src/main/kotlin/org/jetbrains/research/ml/ast/cli/Runner.kt [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+        val yaml by parser.storing(
+            "-y",
+            "--yaml_path",
+            help = "YAML config path"
+        )
+    }
+
+    private fun getTmpProjectDir(): String {
+        val path = "${System.getProperty("java.io.tmpdir")}/tmpProject"
```
Comment (elena-lyulina) [inline src/main/kotlin/org/jetbrains/research/ml/ast/cli/Runner.kt:54]: Maybe we can use some more meaningful name than "tmpProject"? How about "ast-transformations-tmp"?

File: README.md (hunk 3 above)
Comment (elena-lyulina) [reply README.md:34]: It also looks like we apply transformation's n times consistently, like FIle -> Transformed(File) -> Transformed(Transformed(FIle)) and so on. But according to the implementation, we do the first step (FIle -> Transformed(File)) n times. Is it correct?

File: src/main/kotlin/org/jetbrains/research/ml/ast/util/FileUtil.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    if (!file.exists()) {
+        file.mkdirs()
+    }
+}
+
+fun createFile(path: String, content: String = ""): File {
+    val file = File(path)
+    if (file.exists()) {
+        file.delete()
+    }
```
Comment (elena-lyulina) [inline src/main/kotlin/org/jetbrains/research/ml/ast/util/FileUtil.kt:23]: If I understand correctly, `file.writeText(content)` rewrite the context of the file if it's not empty, so there is no need to delete the file, but I may be wrong

**Review by elena-lyulina — CHANGES_REQUESTED** [review]
Cool, thank you

File: src/main/kotlin/org/jetbrains/research/ml/ast/util/FileUtil.kt (hunk 5 above)
Comment (nbirillo) [reply src/main/kotlin/org/jetbrains/research/ml/ast/util/FileUtil.kt:23]: you are right, thanks!

File: README.md [hunk 6]
```diff
@@ -31,14 +31,22 @@ Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-tran
...
 The YAML configuration file example can be found [here](./config.yaml).
-Set of the transformations is applied `n transformations` times. Each round for each transformation the random  
-number between `0.0` and `1.0` is generated. If this value less than `p` for the current transformation,
-the transformation will be applied. The result for each round is stored in the separated folder.
+Set of the transformations is applied `n apply` times. Each round for each transformation the random  
+number between `0.0` and `1.0` is generated. If this value is less than `p` for the current transformation,
+the transformation will be applied to the source files. The result for each round is stored in the separated folder.
+
+For example, the source files are stored in the `source` folder, `n apply` is `5`, and we have list 
+of the `3` transformations. `5` times will be executing the following steps:
```
Comment (elena-lyulina) [inline README.md:39]: we have list of the `3` transformations -> we have a list of `3` transformations

Comment (elena-lyulina) [inline README.md:39]: `5` times will be executing the following steps -> The following steps will be executed `5` times

File: README.md [hunk 7]
```diff
@@ -31,14 +31,22 @@ Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-tran
...
+1. Get list of the transformations for the current round according to the probability.
```
Comment (elena-lyulina) [inline README.md:40]: let's insert **generated**:
according to **generated** probability

File: README.md [hunk 8]
```diff
@@ -31,14 +31,22 @@ Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-tran
...
-number between `0.0` and `1.0` is generated. If this value less than `p` for the current transformation,
-the transformation will be applied. The result for each round is stored in the separated folder.
+Set of the transformations is applied `n apply` times. Each round for each transformation the random  
+number between `0.0` and `1.0` is generated. If this value is less than `p` for the current transformation,
+the transformation will be applied to the source files. The result for each round is stored in the separated folder.
+
+For example, the source files are stored in the `source` folder, `n apply` is `5`, and we have list 
+of the `3` transformations. `5` times will be executing the following steps:
+1. Get list of the transformations for the current round according to the probability. 
+   The size of the list is not more than `3`;
```
Comment (elena-lyulina) [inline README.md:41]: not -> no

File: README.md [hunk 9]
```diff
@@ -31,14 +31,22 @@ Just clone the repo by `git clone https://github.com/JetBrains-Research/ast-tran
...
+2. Apply the list of the transformations to the files from the `source` folder. 
+   **Note**, we create new files to store the result and the files in the `source` folder does not change.
```
Comment (elena-lyulina) [inline README.md:43]: does -> do

File: src/main/kotlin/org/jetbrains/research/ml/ast/cli/Configuration.kt [hunk 10]
```diff
@@ -30,7 +30,7 @@ data class CliTransformation(
 
 @Serializable
 data class Configuration(
-    @SerialName("n transformations")
+    @SerialName("n apply")
     val numTransformations: Int,
```
Comment (elena-lyulina) [inline src/main/kotlin/org/jetbrains/research/ml/ast/cli/Configuration.kt:34]: let's also rename numTransformation, not only its serial name. The same about num

**Review by elena-lyulina — CHANGES_REQUESTED** [review]

**Review by elena-lyulina — APPROVED** [review]
Cool!

**Pull request merged**